app/presenters/corporate_information_page_presenter.rb

```diff
@@ -0,0 +1,69 @@
+class CorporateInformationPagePresenter < ContentItemPresenter
+  include LinkHelper
+
+  def headers_for_contents_list_component
+    return [] unless content_item.headers
+
+    ContentsOutlinePresenter.new(contents_list_headings).for_contents_list_component
+  end
+
+  def corporate_information_heading
+    heading_text = I18n.t("formats.corporate_information_page.corporate_information")
+    heading_id = heading_text.tr(" ", "-").downcase
+
+    {
+      "text" => heading_text,
+      "level" => 2,
+      "id" => heading_id,
+    }
+  end
+
+  def further_information
+    return unless corporate_information_pages?
+
+    further_information_links
+  end
+
+private
+
+  def contents_list_headings
+    content_item.headers << corporate_information_heading if content_item.corporate_information?
+
+    exclude_nested_headings
+
+    ContentsOutline.new(content_item.headers)
+  end
+
+  def exclude_nested_headings
```

Comment:
Is there any restriction on the level? For example, if there are no level 2 headings, and level 3 headings are the top level in the headers hash should the contents list still be displayed?

Reply:
That is a great point, I hadn't thought of that!

I think I'll have to use [this additional check](https://github.com/alphagov/frontend/blob/4af199c3df5f52b86e37b50664c12e847046c0f1/app/models/contents_outline.rb#L10), similar to that in specialist documents - https://github.com/alphagov/frontend/blob/4af199c3df5f52b86e37b50664c12e847046c0f1/app/presenters/specialist_document_presenter.rb#L30

Reply:
I've updated this in commit [Update presenter with contents list headers](https://github.com/alphagov/frontend/pull/4608/commits/e830fee696735aad587a87ca1f08394eaec68a85)